tests/ext/background-sender/agent_headers_unix_domain_socket.phpt

```diff
@@ -0,0 +1,58 @@
+--TEST--
+HTTP headers are sent to the Agent from the background sender over an unix domain socket
+--SKIPIF--
+<?php include __DIR__ . '/../includes/skipif_no_dev_env.inc'; ?>
+--ENV--
+DD_AGENT_HOST=request-replayer
+DD_TRACE_AGENT_URL=unix:///tmp/ddtrace-agent-test.socket
+DD_TRACE_AGENT_FLUSH_AFTER_N_REQUESTS=1
+DD_TRACE_AGENT_FLUSH_INTERVAL=333
+DD_TRACE_GENERATE_ROOT_SPAN=0
+--FILE--
+<?php
+include __DIR__ . '/../includes/request_replayer.inc';
+
+$socketPath = getenv("DD_TRACE_AGENT_URL");
+@unlink($socketPath);
+$p = popen(PHP_BINARY . ' -r \'$s=stream_socket_server("' . $socketPath . '");print 1;$c=stream_socket_accept($s);$t=stream_socket_client("request-replayer:80");$a=$r=[$c,$t];foreach($r as $f)stream_set_blocking($f, false);while(stream_select($r,$w,$e,null)){$d=fread($f=reset($r),4096);if($d=="")exit;fwrite($f==$t?$c:$t,$d);$r=$a;}\'', 'r');
```

Comment:
Must the `-r` value contain no newlines?

Reply:
Can you please suggest a proper formatting? It's just a line to forward proxy data from one socket to the other.

Reply:
I would probably use a heredoc and format it basically as normal PHP, I think? I get that it is "unimportant" but I'd still like to read and understand it :)

Reply:
Also, it appears to be used more than once, so maybe make a function for it?